llvm/lib/CodeGen/MachineOperand.cpp

```diff
@@ -1240,7 +1240,7 @@ void MachineMemOperand::print(raw_ostream &OS, ModuleSlotTracker &MST,
        << "unknown-address";
   }
   MachineOperand::printOperandOffset(OS, getOffset());
-  if (getSize() > 0 && getAlign() != getSize())
+  if (getMemoryType().getSizeInBytes().getKnownMinValue() > 0 && getAlign() != getMemoryType().getSizeInBytes().getKnownMinValue())
```

Comment:
I'm not very confident with this approach. I'm a bit worried if it would this breaks scalar cases.

Reply:
It looks like this check is to emit an `align` operand in the case that the alignment is not the same size as the element size.

That means if the memory type is `nxvMxsN` that we want this guard to be true when `N != Align`. N is the size in bytes of the element type. I think you are checking `M` here.